flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptor.java

```diff
@@ -120,7 +125,7 @@ public TaskDeploymentDescriptor(
 			List<InputGateDeploymentDescriptor> inputGates,
 			List<BlobKey> requiredJarFiles, int targetSlotNumber) {
 
-		this(jobID, vertexID, executionId, taskName, indexInSubtaskGroup, numberOfSubtasks,
+		this(jobID, vertexID, executionId, taskName, indexInSubtaskGroup, numberOfSubtasks, 0,
```

Comment:
This constructor was only being used in test classes, and the attempt number would've been kept 1 for them all. 
But yes. You're right. Leads to more consistency. Will fix this too.